### DISCUSSION TITLE
DNS problem for k3s multicloud cluster

### DISCUSSION THREAD

**Discussion by allnightlong** [k3s-io/k3s#10897] [General]
I building my cluster with nodes from different datacenters. Actually, the cluster has lived  in one dc for some time with 5 node (1 server + 4 agents). Now I'm adding new node in different dc.
Using this tutorial as an example https://docs.k3s.io/networking/distributed-multicloud#embedded-k3s-multicloud-solution

For server:
```
--node-external-ip=<SERVER_EXTERNAL_IP> --flannel-backend=wireguard-native --flannel-external-ip
```
For agent:
```
--node-external-ip=<AGENT_EXTERNAL_IP>
```

The problem is non of the agent's pod can resolve any hostname. I'm using official dns resolution guide https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/ and `nslookup` is failing both for internal and external requests.
Internal:
```
kubectl exec -i -t dnsutils -- nslookup kubernetes.default
;; connection timed out; no servers could be reached


command terminated with exit code 1
```
External:
```
kubectl exec -i -t dnsutils -- nslookup goo.gl                                  
;; connection timed out; no servers could be reached


command terminated with exit code 1
```
External with cloudflare's dns:
```
kubectl exec -i -t dnsutils -- nslookup goo.gl 1.1.1.1
Server:         1.1.1.1
Address:        1.1.1.1#53

Non-authoritative answer:
Name:   goo.gl
Address: 142.250.193.238
Name:   goo.gl
Address: 2404:6800:4002:81d::200e
```

What could be the problem of this DNS issue? How could I resolve it?

P.S. I'm using k3s version `v1.30.4+k3s1` (latest at the time of writing) both for server and agents.

![2024-09-15_02-50-58](https://github.com/user-attachments/assets/54ca6f0d-5e26-40aa-b6d3-c1265b84c182)

**Comment by shakibamoshiri** [comment]
is coredns up and running in **kube-system** namespace ?

example 

```bash
kubectl get pods -n kube-system 
NAME                                      READY   STATUS    RESTARTS   AGE
coredns-576bfc4dc7-w7q9r                  1/1     Running   0          21d
local-path-provisioner-6795b5f9d8-8kvxb   1/1     Running   0          21d
metrics-server-557ff575fb-cqh58           1/1     Running   0          21d
```

if no , first you should check why it does not work , if yes then get the core dns service IP and test it

```bash
kubectl get svc -n kube-system 
NAME             TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                  AGE
kube-dns         ClusterIP   10.13.0.10     <none>        53/UDP,53/TCP,9153/TCP   21d
metrics-server   ClusterIP   10.13.202.13   <none>        443/TCP                  21d
```

test `10.13.0.10` which is coredns service IP

```bash
dig one.com @10.13.0.10
```
